tests/unit/charms/magma_orchestrator_interface/v0/test_magma_orchestrator_interface_requirer.py

```diff
@@ -0,0 +1,106 @@
+    def setUp(self):
+        self.relation_name = "orchestrator"
+        self.model_name = "whatever"
+        self.harness = testing.Harness(DummyMagmaOrchestratorRequirerCharm)
+        self.harness.set_model_name(name=self.model_name)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+        self.maxDiff = None
+
+    @patch(f"{BASE_CHARM_DIR}._on_orchestrator_available")
+    def test_given_orchestrator_information_in_relation_data_when_relation_changed_then_orchestrator_available_event_emitted(  # noqa: E501
+        self, patch_on_orchestrator_available
+    ):
+        remote_app = "magma-orc8r-provider"
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=remote_app
+        )
+
+        root_ca_certificate = "whatever certificate"
+        orchestrator_address = "http://orchestrator.com"
+        orchestrator_port = 123
+        bootstrapper_address = "http://bootstrapper.com"
+        bootstrapper_port = 456
+        fluentd_address = "http://fluentd.com"
+        fluentd_port = 789
+        remote_app_relation_data = {
+            "root_ca_certificate": root_ca_certificate,
+            "orchestrator_address": orchestrator_address,
+            "orchestrator_port": orchestrator_port,
+            "bootstrapper_address": bootstrapper_address,
+            "bootstrapper_port": bootstrapper_port,
+            "fluentd_address": fluentd_address,
+            "fluentd_port": fluentd_port,
+        }
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=remote_app, key_values=remote_app_relation_data
+        )
+
+        patch_on_orchestrator_available.assert_called()
+        args, _ = patch_on_orchestrator_available.call_args
+        orchestrator_available_event = args[0]
+        assert orchestrator_available_event.root_ca_certificate == root_ca_certificate
+        assert orchestrator_available_event.orchestrator_address == orchestrator_address
+        assert orchestrator_available_event.orchestrator_port == orchestrator_port
+        assert orchestrator_available_event.bootstrapper_address == bootstrapper_address
+        assert orchestrator_available_event.bootstrapper_port == bootstrapper_port
+        assert orchestrator_available_event.fluentd_address == fluentd_address
+        assert orchestrator_available_event.fluentd_port == fluentd_port
+
+    @patch(f"{BASE_CHARM_DIR}._on_orchestrator_available")
+    def test_given_orchestrator_information_not_in_relation_data_when_relation_changed_then_orchestrator_available_event_not_emitted(  # noqa: E501
+        self, patch_on_orchestrator_available
+    ):
+        remote_app = "magma-orc8r-provider"
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=remote_app
+        )
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=remote_app, key_values=dict()
```

Comment:
Super nitpicky, still, I was curious if there was any difference between initializing dicts like`dict()` and `{}`? I found this [article](https://katrin-affolter.ch/Programming/Python_dictionaries) and this [post](https://stackoverflow.com/questions/17097985/dict-vs-in-python-which-is-better#:~:text=With%20CPython%202.7%2C%20using%20dict,not%20work%20for%20your%20case.), it seems that is slightly better to use `{}`, although I don't know if it's worth the change.

Reply:
Yeah I don't have a strong opinion here. Personally I'm not a big fan on {} because it seems ambiguous: is it a dict or a set?

Reply:
I ended up moving to {}. I don't want to hassle over this.